bioblend/_tests/TestGalaxyLibraries.py

```diff
@@ -118,3 +118,62 @@ def test_dataset_permissions(self):
         self.assertEqual(set(_[1] for _ in ret_get['access_dataset_roles']), set(user_id_list_new))
         self.assertEqual(set(_[1] for _ in ret_get['modify_item_roles']), set(user_id_list_new))
         self.assertEqual(set(_[1] for _ in ret_get['manage_dataset_roles']), set(user_id_list_new))
+
+    @test_util.skip_unless_galaxy('release_19.09')
+    def test_upload_file_contents_with_tags(self):
+        with tempfile.NamedTemporaryFile(mode='w', prefix='bioblend_test_') as f:
+            f.write(FOO_DATA)
+            f.flush()
+            datasets = self.gi.libraries.upload_file_contents(self.library['id'], FOO_DATA, tags=["foobar", "barfoo"])
+            dataset_show = self.gi.libraries.show_dataset(self.library['id'], datasets[0]['id'])
+            self.assertEqual(dataset_show['tags'], 'name:foobar, name:barfoo')
+
+    @test_util.skip_unless_galaxy('release_19.09')
```

Comment:
Given that all the upload methods just call `_do_upload()`, I think just testing tags with the `test_upload_file_contents_with_tags` test above is enough?
So `test_upload_file_local_path_with_tags` and `test_upload_from_galaxy_filesystem_with_tags` are not needed, IMO.

Reply:
In fact tags are not added the same way between upload_from_galaxy_filesystem and the other methods. I wrote the 3 tests to understand why sometimes the tags were not added.
Now that it works for all, I removed test_upload_file_local_path_with_tags as it's clearly redundant, do you prefer to remove test_upload_from_galaxy_filesystem_with_tags also?

Reply:
I think that tagging is now ensured for all code paths by https://github.com/galaxyproject/galaxy/pull/8530/files#diff-75e29f5ca09c0d8c3d9ef0286e1483dfR284-R286 , so I'd remove also `test_upload_from_galaxy_filesystem_with_tags` .